stratum/hal/lib/barefoot/p4runtime_bfrt_translator.h

```diff
@@ -0,0 +1,165 @@
+// Copyright 2022-present Open Networking Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef STRATUM_HAL_LIB_BAREFOOT_P4RUNTIME_BFRT_TRANSLATOR_H_
+#define STRATUM_HAL_LIB_BAREFOOT_P4RUNTIME_BFRT_TRANSLATOR_H_
+
+#include <map>
+#include <memory>
+#include <string>
+
+#include "absl/memory/memory.h"
+#include "absl/synchronization/mutex.h"
+#include "p4/v1/p4runtime.pb.h"
+#include "stratum/glue/integral_types.h"
+#include "stratum/glue/status/status.h"
+#include "stratum/glue/status/statusor.h"
+#include "stratum/hal/lib/barefoot/bf_sde_interface.h"
+#include "stratum/hal/lib/common/common.pb.h"
+#include "stratum/hal/lib/common/writer_interface.h"
+#include "stratum/lib/macros.h"
+
+namespace stratum {
+namespace hal {
+namespace barefoot {
+
+class P4RuntimeBfrtTranslator {
```

Comment:
Or just `BfrtTranslator`?

Reply:
I'm not sure if this could be a generic P4 component independent from bfrt. So we can do the same for bmv2 or bcm?

Reply:
The `TranslateValue` function is specific for TNA and Barefoot's device

Reply:
Just the `"tna/PortId_t"` string, right?

Reply:
Right, both `TranslateValue` and `TranslateTnaPortId` are specific for TNA/BF

Reply:
Oh, also when chassis config is pushed, we need to collect port mapping from SDE API